orttraining/orttraining/python/training/api/checkpoint_state.py

```diff
@@ -1,19 +1,74 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
-# checkpoint_state.py
+
+from __future__ import annotations
+
+import os
 
 from onnxruntime.capi import _pybind_state as C
 
 
 class CheckpointState:
+    """Class that holds the state of the training session state
+
+    Args:
+        state (CheckpointState): The C.Checkpoint state object that holds the underlying session state.
     """
-    Class for Loading CheckpointState.
-    This class is a wrapper of CheckpointState Class.
-    """
 
-    def __init__(self, ckpt_uri) -> None:
+    def __init__(self, state: C.CheckpointState):
+        if not isinstance(state, C.CheckpointState):
+            raise TypeError(f"Invalid argument for CheckpointState received {type(state)}")
+        self._state = state
+
+    @staticmethod
+    def load_checkpoint(checkpoint_uri: str | os.PathLike) -> CheckpointState:
+        """Loads the checkpoint state from the checkpoint file
+
+        Args:
+            checkpoint_uri: The path to the checkpoint file.
+
+        Returns:
+            CheckpointState: The checkpoint state object.
+        """
+        return CheckpointState(C.load_checkpoint(str(checkpoint_uri)))
+
+    @staticmethod
+    def save_checkpoint(state: CheckpointState, checkpoint_uri: str | os.PathLike) -> None:
+        """Saves the checkpoint state to the checkpoint file
+
+        Args:
+            state: The checkpoint state object.
+            checkpoint_uri: The path to the checkpoint file.
         """
-        Initializes CheckpointState object with the given checkpoint uri.
-        The returned object will be used to initialize the Module.
+        C.save_checkpoint(state._state, str(checkpoint_uri))
```

Comment:
#15530 I just learn about this today

```suggestion
        C.save_checkpoint(state._state, os.fspath(checkpoint_uri))
```